### PR TITLE
PHP8: Code Review `Services/Randomization`

### PR DESCRIPTION
**Pull request by lscharmer**
Hi @mjansenDatabay,

here are the results of the `Services/Randomization` review.

### Results
- [x] The code style is `PSR-2 + X` compliant
- [x] No usage of `$_GET` / `$_POST` / `$_REQUEST` / `$_SESSION`
- [ ] There is a `Unit Test Suite` with at least one unit test
- [ ] The unit tests pass
- [x] External libraries are compatible with PHP 8 (if relevant)
- [x] There are no serious `PHPStorm Code Inspection` issues left
- [x] The types are fully documented (PHPDoc) or explict PHP types are used (type hints, return types, typed properties)

### Changes made in this PR:
- Add Unit Test for class `ilRandom`
- Remove empty `ilRandom` constructor
- Change default values from `null` to `0` and `PHP_INT_MAX` (I checked that there are no usages of `ilRandom::int` where `null` is given as an argument). The max value is changed from `mt_getrandmax` to `PHP_INT_MAX` because `mt_getrandmax` is not relevant for `random_int` only for `mt_rand` and related functions.

Best regards,
@lscharmer

### CODE REVIEW

**Comment by mjansenDatabay** [comment]
Thx a lot @lscharmer 

I am not sure about the first two unit tests. Testing `Random` in general is a philosophical matter we could discuss with a beer and some other community members :-).

I suggest to at least add (e.g.) 1000 iterations, similar to the tests here: https://github.com/ILIAS-eLearning/ILIAS/commit/e277371e7938dff12be993fec785cd8131bbfe00

**Comment by lscharmer** [comment]
> Thx a lot @lscharmer
> 
> I am not sure about the first two unit tests. Testing `Random` in general is a philosophical matter we could discuss with a beer and some other community members :-).
> 
> I suggest to at least add (e.g.) 1000 iterations, similar to the tests here: [e277371](https://github.com/ILIAS-eLearning/ILIAS/commit/e277371e7938dff12be993fec785cd8131bbfe00)

The assertions with `>=`, `<=` and `===` are indeed volatile but they also test if the method can be called without and with arguments without throwing an error and the line `$this->assertTrue($random->int(8, 8) === 8)` tests an edge case to assure that no error is thrown, because the range is expected to be inclusive.

**Comment by lscharmer** [comment]
> Thx a lot @lscharmer
> 
> I am not sure about the first two unit tests. Testing `Random` in general is a philosophical matter we could discuss with a beer and some other community members :-).
> 
> I suggest to at least add (e.g.) 1000 iterations, similar to the tests here: [e277371](https://github.com/ILIAS-eLearning/ILIAS/commit/e277371e7938dff12be993fec785cd8131bbfe00)

I removed the comparison assertions.

File: Services/Randomization/test/ilRandomTest.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        $this->assertInstanceOf(\ilRandom::class, new ilRandom());
+    }
+
+    public function testIntWithoutArguments() : void
+    {
+        $random = new \ilRandom();
+
+        $random->int();
+        $this->assertTrue(true);
+    }
```
Comment (mjansenDatabay) [inline Services/Randomization/test/ilRandomTest.php]: Can we used something like ...

```suggestion
    public function testIntWithoutArguments() : void
    {
        $this->expectNotToPerformAssertions();

        $random = new \ilRandom();

        try {
            $random->int();
        } catch (Error $e) {
            $this->fail('There should be not error when a random integer value is generated');
        }
    }
```

... instead?

File: Services/Randomization/test/ilRandomTest.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        $random->int(-20, 30);
+        $this->assertTrue(true);
+    }
+
+    public function testIntMinimalRange() : void
+    {
+        $random = new \ilRandom();
+
+        $random->int(8, 8);
+        $this->assertTrue(true);
```
Comment (mjansenDatabay) [inline Services/Randomization/test/ilRandomTest.php]: See comment above
